docker.go

```diff
@@ -90,6 +98,37 @@ func (container *Container) EmptyVolumes() *Container {
 	return container
 }
 
+// Define external elements for a volume
+//      name: the name of the volume
+//      provider: the provider of the volume (e.g. dvdi)
+func (v *Volume) ExternalVolume(name, provider string) *ExternalVolume {
+	ev := ExternalVolume{
```

Comment:
nit-pick: You could take the address of `ExternalVolume` directly on the assignment like

```go
ev := &ExternalVolume{
  ...
}
```

and save the address operators below (net win: one `&` less :-) ).